src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -73,13 +93,22 @@ static void HandleNodeBrowse(void * context, DnssdService * services, size_t ser
         discoveryContext->Retain();
         // For some platforms browsed services are already resolved, so verify if resolve is really needed or call resolve callback
 
-        auto & ipAddress = services[i].mAddress;
+        auto & ipAddress         = services[i].mAddress;
+        bool isOperationalBrowse = (strncmp(services[i].mType, kOperationalServiceName, sizeof(kOperationalServiceName)) == 0 &&
+                                    strlen(services[i].mType) == strlen(kOperationalServiceName));
```

Comment:
```suggestion
        bool isOperationalBrowse = (strncmp(services[i].mType, kOperationalServiceName, sizeof(services[i].mType));
```
It doesn't make sense to use `strncmp` with max size set to the length of a string literal (which we know is null-terminated). It's `mType` buffer that needs the bounds check.